lib/archethic/contracts.ex

```diff
@@ -151,24 +210,44 @@ defmodule Archethic.Contracts do
   @spec stop_contract(binary()) :: :ok
   defdelegate stop_contract(address), to: Loader
 
-  @doc """
-  Determines if the contract's execution is valid for the given transaction
-  """
-  @spec valid_contract_execution?(Contract.t(), Transaction.t(), list(Transaction.t())) ::
-          boolean()
-  def valid_contract_execution?(contract = %Contract{}, tx = %Transaction{}, calls \\ []) do
-    case Interpreter.execute(:transaction, contract, tx, [tx | calls]) do
-      {:ok, _} ->
-        true
-
-      _ ->
-        false
-    end
-  end
-
   @doc """
   Returns a contract instance from a transaction
   """
   @spec from_transaction(Transaction.t()) :: {:ok, Contract.t()} | {:error, String.t()}
   defdelegate from_transaction(tx), to: Contract, as: :from_transaction
+
+  defp get_condition_constants(
+         :inherit,
+         %Contract{constants: %Constants{contract: contract_constant}},
+         transaction,
+         datetime
+       ) do
+    %{
+      "previous" => contract_constant,
+      "next" => Constants.from_transaction(transaction),
+      "_time_now" => DateTime.to_unix(datetime)
+    }
+  end
+
+  defp get_condition_constants(
+         _,
+         %Contract{constants: %Constants{contract: contract_constant}},
+         transaction,
+         datetime
+       ) do
+    %{
+      "transaction" => Constants.from_transaction(transaction),
+      "contract" => contract_constant,
+      "_time_now" => DateTime.to_unix(datetime)
+    }
+  end
+
+  defp get_condition(:oracle, %Contract{conditions: conditions}),
```

Comment:
Don't you think this can be simplied over
```elixir
defp get_condition(condition, %Contract{conditions: conditions}) when condition in [:oracle, :transaction, :inherit] do
    Map.get(conditions, condition)
end
```